templates/repo/diff/box.tmpl

```diff
@@ -166,6 +166,9 @@
 										<a class="ui basic tiny button" rel="nofollow" href="{{$.BeforeSourcePath}}/{{PathEscapeSegments .Name}}">{{ctx.Locale.Tr "repo.diff.view_file"}}</a>
 									{{else}}
 										<a class="ui basic tiny button" rel="nofollow" href="{{$.SourcePath}}/{{PathEscapeSegments .Name}}">{{ctx.Locale.Tr "repo.diff.view_file"}}</a>
+										{{if and $.Repository.CanEnableEditor $.CanEditFile (not $file.IsLFSFile) (not $file.IsBin)}}
+											<a class="ui basic tiny button" rel="nofollow" href="{{$.HeadRepoLink}}/_edit/{{PathEscapeSegments $.HeadBranchName}}/{{PathEscapeSegments $file.Name}}?hide_create_pr=true&hide_create_new_branch=true&redirect_to={{$.BackToLink}}{{PathEscape "#diff-"}}{{$file.NameHash}}">{{ctx.Locale.Tr "repo.editor.edit_this_file"}}</a>
```

Comment:
I think it needs this:  Improve QueryEscape helper function #29768 

Then it doesn't need such strange Escape/Unescape anymore.

Reply:
With  #29768, it could be :

```
redirect_to={{print $.BackToLink "#diff-" $file.NameHash | QueryEscape}}
```

And backend code doesn't need to to double-unescape

Reply:
afe9aaf